internal/datasource/state/resource.go

```diff
@@ -0,0 +1,97 @@
+package custom_type
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/labd/commercetools-go-sdk/platform"
+
+	"github.com/labd/terraform-provider-commercetools/internal/utils"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &StateSource{}
+	_ datasource.DataSourceWithConfigure = &StateSource{}
+)
+
+// NewDataSource is a helper function to simplify the data source implementation.
+func NewDataSource() datasource.DataSource {
+	return &StateSource{}
+}
+
+// StateSource is the data source implementation.
+type StateSource struct {
+	client *platform.ByProjectKeyRequestBuilder
+	mutex  *utils.MutexKV
```

Comment:
Do we need a mutex for reading? In the resource we have the mutex because we have split two resources using the same commercetools endpoint. But for reading it might not be needed

Reply:
No I don't think so. Its a pure read action, so there is no real chance of a race condition happening here.